app/backend/approaches/tabulardataassistant.py

```diff
@@ -0,0 +1,235 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import base64
+import os
+import glob
+import warnings
+import re
+import pandas as pd
+from langchain.chat_models import ChatOpenAI
+from langchain_experimental.agents.agent_toolkits import create_pandas_dataframe_agent
+from langchain.agents.agent_types import AgentType
+from langchain.chat_models import AzureChatOpenAI
+from langchain.agents import load_tools
+import matplotlib.pyplot as plt
+
+warnings.filterwarnings('ignore')
+from dotenv import load_dotenv
+
+# Initialize session state
+# if 'show_images' not in st.session_state:
+#     st.session_state.show_images = False
+
+#--------------------------------------------------------------------------
+#variables needed for testing
+OPENAI_API_TYPE = "azure"
+OPENAI_API_VERSION = "2023-06-01-preview"
+OPENAI_API_BASE = " "
+OPENAI_API_KEY = " "
+OPENAI_DEPLOYMENT_NAME = " "
+MODEL_NAME = " "
+AZURE_OPENAI_ENDPOINT = ' '
+AZURE_OPENAI_SERVICE_KEY = ' '
+
+os.environ["OPENAI_API_TYPE"] = OPENAI_API_TYPE
+os.environ["OPENAI_API_VERSION"] = OPENAI_API_VERSION
+#os.environ["OPENAI_API_BASE"] = OPENAI_API_BASE
```

Comment:
probbaly good idea to clean the entire code by removing all commented code. You can save it for your refrence in your copy but here it is making it hard to read the entire code

Reply:
updated